server/chat-plugins/mafia.js

```diff
@@ -535,10 +535,10 @@ class MafiaTracker extends Rooms.RoomGame {
 		let roles = Dex.shuffle(this.roles.slice());
 		if (roles.length) {
 			for (let p in this.players) {
-				let role = roles.shift();
+				let role = roles.shift() || null;
```

Comment:
Is there any chance this will actually be `null`? Either way, it seems the correct choice is to do the cast here rather than lower down.

Reply:
I originally cast here, assuming rules is always at least as long as player, but then I noticed `Player.role` is `MafiaRole?` and so could be null (though maybe its only supposed to be null for initialization?). Fine with casting here though, its less ugly for sure.

Reply:
The lack of check lower down implies that this is not expected to ever be null, though. Mafia isn't my code so I don't know for sure what the intent is, of course.

Reply:
Not quite. The lack of check lower down is under `if (u && u.connected)`, so its only implied to not be null in that circumstance, hence the original patch (I didnt look into what `Users` function was doing with player, but it could conceivably modify `.role` in the case that it was null, or not consider the user `connected` if `role` is null, etc). I think its fine to have moved the cast though.